src/policy/satisfy.rs

```diff
@@ -196,7 +217,94 @@ impl<Pk: MiniscriptKey + PublicKey32 + ToPublicKey> Policy<Pk> {
                     Some(right)
                 }
             }
-            Policy::Threshold(_, _) => None,
+            Policy::Threshold(k, sub_policies) => {
+                assert!(
+                    sub_policies.len() >= 2,
+                    "Thresholds must have at least two sub-policies"
+                );
+
+                let mut satisfiable_children = Vec::new();
+
+                for (index, policy) in sub_policies.iter().enumerate() {
+                    if let Some(ext) = policy.satisfy_helper(satisfier, context) {
+                        satisfiable_children.push((ext, index));
+                    }
+                }
+
+                if satisfiable_children.len() < *k {
+                    return None;
+                }
+
+                // 1) Sort by witness cost
+                // 2) Select k best satisfactions
+                // 3) Sort by index in decreasing order
+                satisfiable_children.sort();
```

Comment:
In 027ab50b45d65b555086036e6ccd588679013976:

Nit: using `sort_by_key` would be faster and avoid the need to implement `Ord`/`PartialOrd` on `SatisfierExtData`.